util/DHI.Mike1D.MikeIO/ResultDataCopier.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using DHI.Mike1D.ResultDataAccess;
+
+namespace DHI.Mike1D.MikeIO
+{
+  /// <summary>
+  /// Class which copies in the result data time series into
+  /// an allocated memory specified by a pointer.
+  /// </summary>
+  public class ResultDataCopier
+  {
+    /// <summary>
+    /// Instance of ResultData object to work with.
+    /// </summary>
+    public ResultData ResultData { get => _resultData; set => _resultData = value; }
+    private ResultData _resultData;
+
+    /// <inheritdoc />
+    public ResultDataCopier(ResultData resultData)
+    {
+      _resultData = resultData;
+    }
+
+    /// <summary>
+    /// Creates and empty list for DataEntry objects.
+    /// </summary>
+    public List<DataEntry> GetEmptyDataEntriesList()
+    {
+      return new List<DataEntry>();
+    }
+
+    /// <summary>
+    /// Copies all the ResultData data items into memory specified by a given pointer.
+    /// </summary
+    public void CopyData(IntPtr intPointer)
+    {
+      var dataEntries = GetAllDataEntries();
+      CopyData(intPointer, dataEntries);
+    }
+
+    /// <summary>
+    /// Creates a list of all data entries corresponding to all data items.
+    /// </summary
+    public List<DataEntry> GetAllDataEntries()
+    {
+      var dataEntries = new List<DataEntry>();
```

Comment:
I prefer Linq, which I think is more readable, but if this the convention, go ahead.
```csharp
using System.Linq;

...

 var dataEntries = _resultData.DataSets
        .SelectMany(dataSet => dataSet.DataItems)
        .SelectMany(dataItem => Enumerable.Range(0, dataItem.NumberOfElements)
        .Select(elementIndex => new DataEntry(dataItem, elementIndex)))
        .ToList();
      return dataEntries;
````

Reply:
I agree that Linq is more readable and it is also a good place for Linq, because such method call is not frequent. Though, in MIKE 1D we typically tend to avoid Linq statements for performance reasons (relevant for current .NET 4.7 framework used by MIKE software).